app/src/lib/git/rev-parse.ts

```diff
@@ -52,6 +52,32 @@ export async function getTopLevelWorkingDirectory(
   return Path.resolve(path, relativePath)
 }
 
+export async function checkIfRepositoryIsBare(
+  path: string
+): Promise<boolean | null> {
+  let result
+
+  try {
+    result = await git(
+      ['rev-parse', '--is-bare-repository'],
+      path,
+      'checkIfRepositoryIsBare'
```

Comment:
:art: this should match the function being called

Reply:
This `'checkIfRepositoryIsBare'` string is used for tracing Git operations (in case they run unexpectedly long or if they error, details end up in the log). The convention is that this name should be the same as the function itself.

Reply:
The source of my confusion is that the function I have added is named checkIfRepositoryIsBare and I am passing a string 'checkIfRepositoryIsBare'.

Similar to how getTopLevelWorkingDirectory passes 'getTopLevelWorkingDirectory' as an argument.

Did I make a mistake with the string? One thing I noticed that seems relevant is that there is 'isGitRepostory' and 'getTopLevelWorkingDirectory', with the prior calling the latter. I did not copy this pattern and only have checkIfRepostoryIsBare. Perhaps that is part of my problem.

Sorry to take up your time with this.

Reply:
We can't follow this pattern for `isGitRepository` because it defers to `getTopLevelWorkingDirectory` for it's Git operation.

The pattern we're trying to follow here is **if a function performs a Git operation, ensure it's named in a way that we can trace back to later**.

What I was looking to ensure here is that the function name itself is in sync with the string we pass into `git()`, and I think we're on the same page. I just wanted to ensure the function name change mentioned above https://github.com/desktop/desktop/pull/5434#discussion_r213014336 is also applied here.